File: php-pr/51/php/code-quality-error-handling/FileReader.php
<?php

class FileReader
{
    public function readFile($filename)
    {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        try {
            $content = file_get_contents($filename);
            return $content;
        } catch (Exception $e) {
            return false; // Hiding exception details
        }
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-error-handling/file_reader.rb
class FileReader
  def read_file(filename)
    # {fact rule=code-quality-error-handling@v1.0 defects=1}
    begin
      content = File.read(filename)
      return content
    rescue => e
      return false # Hiding exception details
    end
    # {/fact}
  end
end